Compilation of solo5-kernel-ukvm 0.3.0 fails

On Fedora 28, gcc 8.1.1 and ocaml 4.04.2 compilation of solo5-kernel-ukvm fails with the following error:

```
$ opam install solo5-kernel-ukvm
The following actions will be performed:
  ∗  install solo5-kernel-ukvm 0.3.0

=-=- Gathering sources =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
[solo5-kernel-ukvm] Archive in cache

=-=- Processing actions -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
[ERROR] The compilation of solo5-kernel-ukvm failed at "make ukvm".
Processing  1/1: [solo5-kernel-ukvm: make opam-ukvm-uninstall]
#=== ERROR while installing solo5-kernel-ukvm.0.3.0 ===========================#
# opam-version 1.2.2
# os           linux
# command      make ukvm
# path         /home/svanders/.opam/4.04.2/build/solo5-kernel-ukvm.0.3.0
# compiler     4.04.2
# exit-code    2
# env-file     /home/svanders/.opam/4.04.2/build/solo5-kernel-ukvm.0.3.0/solo5-kernel-ukvm-18863-1872ec.env
# stdout-file  /home/svanders/.opam/4.04.2/build/solo5-kernel-ukvm.0.3.0/solo5-kernel-ukvm-18863-1872ec.out
# stderr-file  /home/svanders/.opam/4.04.2/build/solo5-kernel-ukvm.0.3.0/solo5-kernel-ukvm-18863-1872ec.err
### stdout ###
# [...]
# cc -nostdinc -fno-stack-protector -ffreestanding -mno-red-zone -mno-sse -mno-mmx -mno-aes -mno-avx -isystem /home/svanders/.opam/4.04.2/build/solo5-kernel-ukvm.0.3.0/include-host -std=gnu99 -Wall -Wextra -Werror -O2 -g -D__SOLO5_KERNEL__ -c ukvm/net.c -o ukvm/net.o
# cc -nostdinc -fno-stack-protector -ffreestanding -mno-red-zone -mno-sse -mno-mmx -mno-aes -mno-avx -isystem /home/svanders/.opam/4.04.2/build/solo5-kernel-ukvm.0.3.0/include-host -std=gnu99 -Wall -Wextra -Werror -O2 -g -D__SOLO5_KERNEL__ -c ukvm/block.c -o ukvm/block.o
# cc -nostdinc -fno-stack-protector -ffreestanding -mno-red-zone -mno-sse -mno-mmx -mno-aes -mno-avx -isystem /home/svanders/.opam/4.04.2/build/solo5-kernel-ukvm.0.3.0/include-host -std=gnu99 -Wall -Wextra -Werror -O2 -g -D__SOLO5_KERNEL__ -c abort.c -o abort.o
# cc -nostdinc -fno-stack-protector -ffreestanding -mno-red-zone -mno-sse -mno-mmx -mno-aes -mno-avx -isystem /home/svanders/.opam/4.04.2/build/solo5-kernel-ukvm.0.3.0/include-host -std=gnu99 -Wall -Wextra -Werror -O2 -g -D__SOLO5_KERNEL__ -c cpu_x86_64.c -o cpu_x86_64.o
# cc -nostdinc -fno-stack-protector -ffreestanding -mno-red-zone -mno-sse -mno-mmx -mno-aes -mno-avx -isystem /home/svanders/.opam/4.04.2/build/solo5-kernel-ukvm.0.3.0/include-host -std=gnu99 -Wall -Wextra -Werror -O2 -g -D__SOLO5_KERNEL__ -DASM_FILE -c cpu_vectors_x86_64.S -o cpu_vectors_x86_64.o
# cc -nostdinc -fno-stack-protector -ffreestanding -mno-red-zone -mno-sse -mno-mmx -mno-aes -mno-avx -isystem /home/svanders/.opam/4.04.2/build/solo5-kernel-ukvm.0.3.0/include-host -std=gnu99 -Wall -Wextra -Werror -O2 -g -D__SOLO5_KERNEL__ -c ee_printf.c -o ee_printf.o
# cc -nostdinc -fno-stack-protector -ffreestanding -mno-red-zone -mno-sse -mno-mmx -mno-aes -mno-avx -isystem /home/svanders/.opam/4.04.2/build/solo5-kernel-ukvm.0.3.0/include-host -std=gnu99 -Wall -Wextra -Werror -O2 -g -D__SOLO5_KERNEL__ -c intr.c -o intr.o
# cc -nostdinc -fno-stack-protector -ffreestanding -mno-red-zone -mno-sse -mno-mmx -mno-aes -mno-avx -isystem /home/svanders/.opam/4.04.2/build/solo5-kernel-ukvm.0.3.0/include-host -std=gnu99 -Wall -Wextra -Werror -O2 -g -D__SOLO5_KERNEL__ -c lib.c -o lib.o
# cc -nostdinc -fno-stack-protector -ffreestanding -mno-red-zone -mno-sse -mno-mmx -mno-aes -mno-avx -isystem /home/svanders/.opam/4.04.2/build/solo5-kernel-ukvm.0.3.0/include-host -std=gnu99 -Wall -Wextra -Werror -O2 -g -D__SOLO5_KERNEL__ -c mem.c -o mem.o
# make[1]: Leaving directory '/home/svanders/.opam/4.04.2/build/solo5-kernel-ukvm.0.3.0/kernel'
### stderr ###
# mem.c:58:5: error: array subscript -1 is below array bounds of ‘char[]’ [-Werror=array-bounds]
# [...]
# mem.c:59:5: error: array subscript -1 is below array bounds of ‘char[]’ [-Werror=array-bounds]
#      log(INFO, "Solo5:     rodata @ (0x%lx - 0x%lx)\n", &_etext, &_erodata[-1]);
#      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# mem.c:60:5: error: array subscript -1 is below array bounds of ‘char[]’ [-Werror=array-bounds]
#      log(INFO, "Solo5:       data @ (0x%lx - 0x%lx)\n", &_erodata, &_end[-1]);
#      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# cc1: all warnings being treated as errors
# make[1]: *** [GNUmakefile:98: mem.o] Error 1
# make: *** [GNUmakefile:39: ukvm] Error 2



=-=- Error report -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
The following actions failed
  ∗  install solo5-kernel-ukvm 0.3.0
No changes have been performed

=-=- solo5-kernel-ukvm.0.3.0 troobleshooting =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
=> This package relies on external (system) dependencies that may be missing.
   `opam depext solo5-kernel-ukvm.0.3.0' may help you find the correct
   installation for your system.

```

gcc version:
```
$ gcc --version
gcc (GCC) 8.1.1 20180502 (Red Hat 8.1.1-1)
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

--- a/kernel/mem.c
+++ b/kernel/mem.c
@@ -54,10 +54,10 @@ void mem_init(void)
 	PANIC("Not enough memory", NULL);
 
     log(INFO, "Solo5: Memory map: %lu MB addressable:\n", mem_size >> 20);
-    log(INFO, "Solo5:     unused @ (0x0 - 0x%lx)\n", &_stext[-1]);
-    log(INFO, "Solo5:       text @ (0x%lx - 0x%lx)\n", &_stext, &_etext[-1]);
-    log(INFO, "Solo5:     rodata @ (0x%lx - 0x%lx)\n", &_etext, &_erodata[-1]);
-    log(INFO, "Solo5:       data @ (0x%lx - 0x%lx)\n", &_erodata, &_end[-1]);
+    log(INFO, "Solo5:     unused @ (0x0 - 0x%lx)\n", _stext-1);
+    log(INFO, "Solo5:       text @ (0x%lx - 0x%lx)\n", _stext, _etext-1);
+    log(INFO, "Solo5:     rodata @ (0x%lx - 0x%lx)\n", _etext, _erodata-1);
+    log(INFO, "Solo5:       data @ (0x%lx - 0x%lx)\n", _erodata, _end-1);
     log(INFO, "Solo5:       heap >= 0x%lx < stack < 0x%lx\n", heap_start,
         mem_size);
 }

